components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/core/CarbonAxisConfigurator.java

```diff
@@ -385,30 +387,26 @@ public void setRepoUpdateFailed() {
 
     @Override
     protected void startSearch(RepositoryListener listener) {
-        //TODO need to implement
- /*       schedulerTask = new CarbonDeploymentSchedulerTask(listener,
-                                                          axisConfig,
-                                                          MultitenantConstants.SUPER_TENANT_ID,
-                                                          MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
-        scheduler = Executors
-                .newScheduledThreadPool(1, new CarbonThreadFactory(new ThreadGroup("CarbonDeploymentSchedulerThread")));
-        String deploymentInterval =
-                CoreServerInitializer.
-                        getServerConfigurationService().getFirstProperty("Axis2Config.DeploymentUpdateInterval");
+        CarbonDeploymentSchedulerTask schedulerTask = new CarbonDeploymentSchedulerTask(listener, axisConfig);
+        scheduler = Executors.newScheduledThreadPool(1, new CarbonThreadFactory(
+                new ThreadGroup("CarbonDeploymentSchedulerThread")));
+
+        String deploymentInterval = CarbonCoreDataHolder.getInstance().getServerConfigurationService().getFirstProperty(
```

Comment:
It seems like this configuration is not present in the axis2.xml. Shall we remove this block of code and create a separate issue to make the value configurable?

Reply:
This is in carbon xml but are we going to fetch from there or change to toml ?

Reply:
This configuration is being read from the carbon.xml file (not from the axis2.xml file) and it is present in there as a configurable element.